packages/client/src/routings.ts

```diff
@@ -0,0 +1,111 @@
+import { type ContentRouting } from '@libp2p/interface/content-routing'
+import { CodeError } from '@libp2p/interface/errors'
+import { type PeerRouting } from '@libp2p/interface/peer-routing'
+import { peerIdFromBytes } from '@libp2p/peer-id'
+import { marshal, unmarshal } from 'ipns'
+import first from 'it-first'
+import map from 'it-map'
+import { equals as uint8ArrayEquals } from 'uint8arrays/equals'
+import { fromString as uint8ArrayFromString } from 'uint8arrays/from-string'
+import type { DelegatedRoutingV1HttpApiClient } from './index.js'
+import type { AbortOptions } from '@libp2p/interface'
+import type { PeerId } from '@libp2p/interface/peer-id'
+import type { PeerInfo } from '@libp2p/interface/peer-info'
+import type { CID } from 'multiformats/cid'
+
+const IPNS_PREFIX = uint8ArrayFromString('/ipns/')
+
+function isIPNSKey (key: Uint8Array): boolean {
+  return uint8ArrayEquals(key.subarray(0, IPNS_PREFIX.byteLength), IPNS_PREFIX)
+}
+
+const peerIdFromRoutingKey = (key: Uint8Array): PeerId => {
+  return peerIdFromBytes(key.slice(IPNS_PREFIX.length))
+}
+
+/**
+ * Wrapper class to convert events into returned values
+ */
```

Comment:
```suggestion
/**
 * Wrapper class to convert [http-routing-v1 content events](https://specs.ipfs.tech/routing/http-routing-v1/#response-body) into returned values
 */
```

or relevant jsdoc `{@link}` to TypeScript type that links to source spec.